exchanges/poloniex/poloniex_types.go

```diff
@@ -505,16 +480,545 @@ type WalletWithdrawals struct {
+	AmendedType          string  `json:"type,omitempty,string"`
+	TimeInForce          string  `json:"timeInForce,omitempty"`
+	ProceedOnFailure     bool    `json:"proceedOnFailure,omitempty,string"` // proceedOnFailure flag is intended to specify whether to continue with new smart order placement in case cancellation of the existing smart order fails.
+}
+
+// CancelReplaceSmartOrder represents a response parameter for order cancellation and replacement operation.
+type CancelReplaceSmartOrder struct {
+	ID        string       `json:"id"`
+	StopPrice types.Number `json:"stopPrice"`
+	Price     types.Number `json:"price"`
+	Quantity  types.Number `json:"quantity"`
+	Code      int64        `json:"code"`
+	Message   string       `json:"message"`
+}
+
+// SmartOrderDetails represents a smart order information and trigger detailed information.
+type SmartOrderDetails struct {
+	ID             string            `json:"id"`
+	ClientOrderID  string            `json:"clientOrderId"`
+	Symbol         string            `json:"symbol"`
+	State          string            `json:"state"`
+	AccountType    string            `json:"accountType"`
+	Side           string            `json:"side"`
+	Type           string            `json:"type"`
+	TimeInForce    order.TimeInForce `json:"timeInForce"`
+	Quantity       types.Number      `json:"quantity"`
+	Price          types.Number      `json:"price"`
+	Amount         types.Number      `json:"amount"`
+	StopPrice      types.Number      `json:"stopPrice"`
+	CreateTime     types.Time        `json:"createTime"`
+	UpdateTime     types.Time        `json:"updateTime"`
+	TriggeredOrder TradeOrder        `json:"triggeredOrder"`
+}
+
+// CancelSmartOrderResponse represents a close cancel smart order response instance.
+type CancelSmartOrderResponse struct {
+	OrderID       string `json:"orderId"`
+	ClientOrderID string `json:"clientOrderId"`
+	State         string `json:"state"`
+	Code          int64  `json:"code"`
+	Message       string `json:"message"`
+}
+
+// TradeHistory represents an order trade history instance.
+type TradeHistory struct {
+	ID            string        `json:"id"`
+	ClientOrderID string        `json:"clientOrderId"`
+	Symbol        string        `json:"symbol"`
+	AccountType   string        `json:"accountType"`
+	OrderID       string        `json:"orderId"`
+	Side          string        `json:"side"`
+	Type          string        `json:"type"`
+	MatchRole     string        `json:"matchRole"`
+	Price         types.Number  `json:"price"`
+	Quantity      types.Number  `json:"quantity"`
+	Amount        types.Number  `json:"amount"`
+	FeeCurrency   currency.Code `json:"feeCurrency"`
+	FeeAmount     types.Number  `json:"feeAmount"`
+	PageID        string        `json:"pageId"`
+	CreateTime    types.Time    `json:"createTime"`
+}
+
+// SubscriptionPayload represents a subscriptions request instance structure.
+type SubscriptionPayload struct {
+	ID         string         `json:"id"`
```

Comment:
```suggestion
	ID         string         `json:"id,omitempty"`
```

In the normal subs we are sending out ID which is never used or passed back